airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py

```diff
@@ -26,9 +26,84 @@
 from pipelines.dagger.actions import secrets
 from pipelines.dagger.actions.python.poetry import with_poetry
 from pipelines.helpers.utils import METADATA_FILE_NAME, get_exec_result
+from pipelines.models.secrets import Secret, SecretNotFoundError, SecretStore
 from pipelines.models.steps import STEP_PARAMS, MountPath, Step, StepResult, StepStatus
 
 
+def get_secrets_from_connector_test_suites_option(
+    connector_test_suites_options: List[Dict[str, str | Dict[str, List[Dict[str, str | Dict[str, str]]]]]],
+    suite_name: str,
+    secret_stores: Dict[str, SecretStore],
+    raise_on_missing_secret_store: bool = True,
+    logger: logging.Logger | None = None,
+) -> List[Secret]:
+    """Get secrets declared in metadata connectorTestSuitesOptions for a test suite name.
+    It will use the secret store alias declared in connectorTestSuitesOptions.
+    If the secret store is not available a warning or and error could be raised according to the raise_on_missing_secret_store parameter value.
+    We usually want to raise an error when running in CI context and log a warning when running locally, as locally we can fallback on local secrets.
+
+    Args:
+        connector_test_suites_options (List[Dict[str, str  |  Dict]]): The connector under test test suite options
+        suite_name (str): The test suite name
+        secret_stores (Dict[str, SecretStore]): The available secrets stores
+        raise_on_missing_secret_store (bool, optional): Raise an error if the secret store declared in the connectorTestSuitesOptions is not available. Defaults to True.
+        logger (logging.Logger | None, optional): Logger to log a warning if the secret store declared in the connectorTestSuitesOptions is not available. Defaults to None.
+
+    Raises:
+        SecretNotFoundError: Raised  if the secret store declared in the connectorTestSuitesOptions is not available and raise_on_missing_secret_store is truthy.
+
+    Returns:
+        List[Secret]: List of secrets declared in the connectorTestSuitesOptions for a test suite name.
+    """
+    secrets: List[Secret] = []
+
+    for enabled_test_suite in connector_test_suites_options:
+        if enabled_test_suite["suite"] == suite_name:
+            if enabled_test_suite.get("testSecrets"):
+                assert isinstance(enabled_test_suite["testSecrets"], list)
+                suite_secrets: List[Dict[str, str | Dict[str, str]]] = enabled_test_suite["testSecrets"]
+                for s in suite_secrets:
+                    if s["secretStore"]["alias"] not in secret_stores:
+                        message = f"Secret {s['name']} can't be retrieved as {s['secretStore']['alias']} is not available"
+                        if raise_on_missing_secret_store:
+                            raise SecretNotFoundError(message)
+                        if logger is not None:
+                            logger.warn(message)
```

Comment:
Under what circumstances would it be okay to declare a secret store but have it not be available? This will just error downstream otherwise, right? (In a possibly unpredictable way that might not have a nice error message.)

It feels like we should always raise if the secret was declared but not available (and then we can get rid of the raise_on_missing_secret_store code paths).

Reply:
@clnoll when the users does not have access to our GSM secret store it won't be available.
`get_connector_secrets_for_test_suite`  does the following:
* Fetch secrets from remote secret store by calling `get_secrets_from_connector_test_suites_option`
* Fetch secrets from the local secret store (in the `secrets` connector directory)

Reply:
But in this case the secret store is the local secret store instead of the GSM one, right? In which case there will still be a secret store available?

Reply:
> But in this case the secret store is the local secret store instead of the GSM one, right?

Secrets can come from various secret store.
The current logic concatenates secrets from GSM and local ones.